homeassistant/components/nut/device_action.py

```diff
@@ -0,0 +1,81 @@
+"""Provides device actions for Network UPS Tools (NUT)."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_TYPE
+from homeassistant.core import Context, HomeAssistant
+from homeassistant.helpers import device_registry
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType, TemplateVarsType
+
+from . import PyNUTData
+from .const import (
+    DEVICE_SUPPORTED_COMMANDS,
+    DOMAIN,
+    INTEGRATION_SUPPORTED_COMMANDS,
+    PYNUT_DATA,
+)
+
+ACTION_TYPES = {cmd.replace(".", "_") for cmd in INTEGRATION_SUPPORTED_COMMANDS}
+
+ACTION_SCHEMA = cv.DEVICE_ACTION_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_TYPE): vol.In(ACTION_TYPES),
+    }
+)
+
+
+async def async_get_actions(
+    hass: HomeAssistant, device_id: str
+) -> list[dict[str, str]]:
+    """List device actions for Network UPS Tools (NUT) devices."""
+    entry_id = _get_entry_id_from_device_id(hass, device_id)
+    if entry_id is None:
+        return []
+
+    actions = []
+    supported_commands: set[str] = hass.data[DOMAIN][entry_id][
+        DEVICE_SUPPORTED_COMMANDS
+    ]
+    for command_name in supported_commands:
+        device_action_name = _get_device_action_name(command_name)
+
+        base_action = {
+            CONF_DEVICE_ID: device_id,
+            CONF_DOMAIN: DOMAIN,
+        }
```

Comment:
This never changes so it should move outside of the loop above

Reply:
Great catch! I had copied it from somewhere else but I had not realized that after I tweaked it to not include the entry then it could be moved outside.